File: ts/src/types/retry.ts
/**
 * Retry behavior contract for transport/protocol operations.
 */
export interface RetryPolicy {
  /**
   * Enables/disables retry behavior.
   */
  enabled: boolean;

  /**
   * Total number of attempts, including the first request.
   */
  maxAttempts: number;

  /**
   * Initial backoff delay in milliseconds.
   */
  baseDelayMs: number;

  /**
   * Maximum backoff delay in milliseconds.
   */
  maxDelayMs: number;

  /**
   * Enables jitter for backoff delays.
   */
  jitter: boolean;

  /**
   * Additional HTTP status codes to treat as retryable.
   */
  retryableStatusCodes?: number[];
}

/**
 * Partial retry override used in client config and per-call options.
 */
export type RetryPolicyOverride = Partial<RetryPolicy>;

File: py/src/aembit_edge/retry.py
"""Public retry configuration types."""

from dataclasses import dataclass


@dataclass(slots=True, kw_only=True)
class RetryPolicy:
    """Retry policy override for transport requests."""

    enabled: bool | None = None
    max_attempts: int | None = None
    base_delay_ms: int | None = None
    max_delay_ms: int | None = None
    retry_on_status_codes: tuple[int, ...] | None = None
